transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -404,92 +406,221 @@ static jint netty_epoll_native_tcpMd5SigMaxKeyLen(JNIEnv* env, jclass clazz) {
 
 //LibAIO methods
 
-static jlong netty_epoll_native_createAIOContext0(JNIEnv* env, jclass clazz, jint concurrency)
+// the maximum number of buffers for a vectored IO request
+#define MAX_NUM_BUFFERS_PER_REQUEST 8
```

Comment:
We set a maximum number of buffers to avoid allocations and because the kernel code is also optimized to use inline arrays for less than 8 buffers. However, it's probably not that bad to remove this maximum and allocate arrays dynamically, if this is a concern (note the array in `netty_iocb`).